openstack/CleanWlmDatabase/historical_data_cleanup/DeleteHistoricalRecords.py

```diff
@@ -0,0 +1,88 @@
+import os
+import configparser
+import re
+import mysql.connector
+
+# Get records to be deleted using select query
+# Write those records into file
+# Run delete on those records
+
+batchSize=1000
+
+# DB connection 
+## Grab fields from the conf file
+config = configparser.ConfigParser()
+config.read('/etc/workloadmgr/workloadmgr.conf')
+mysql_conn = config['DEFAULT']['sql_connection']
+db_string = re.split('mysql://|=|@|/|:|\?', mysql_conn)
+
+## Edit these 4 variable manually if you want to change any of them
+db_user = db_string[1]
+db_pass = db_string[2]
+db_host = db_string[3]
+db_db = db_string[4]
+
+my_connect = mysql.connector.connect(user=db_user, password=db_pass, host=db_host, database=db_db)
+cursor = my_connect.cursor(dictionary=True)
+
+
+def getRecords(batchSize):
+    try:
+       filepath="./select_queries/{}".format(table)
+       tablename = open(filepath,"r")
+       #print(tablename)
+       query=tablename.readline()
+       tablename.close()
+       record_count=1
+       while record_count > 0:
+          print('\n\nGetting records from {}'.format(table))
+          query=query.replace("#LIMIT#",str(batchSize))
+          print("Running query : {}".format(query))
+          cursor.execute(query)
+          result=cursor.fetchall()
+          record_count=len(result)
+          writeRecords(result)
+    except IOError:
+       print('File for table {} not present'.format(table))
+
+def writeRecords(data):
+    print('\nWritting records to be deleted from table {}'.format(table))
+    try:
+       filepath="./recordsToBeDeleted/{}".format(table)
+       f = open(filepath,"w")
+       for x in data:
+         for i in x.values(): 
+            f.writelines(i+"\n")
+            #print(i)
+       f.close()
+       deleteRecords()
```

Comment:
In previous utilities, required records were directly deleted and if any timeout happened, then user was expected to re execute the utility. In such case, records processed until timeout were already removed from database.
In current flow of doing select first and then delete, are we loosing that advantage?
And is the total time required for select + delete is less than direct delete?

Reply:
Yes in this approach we will loose that advantage of intermediate delete. But we will have less time of lock on table. As this procedure will be long running, from customer perspective its better to have less lock time. 
Select + Delete approach takes less time that previous utility. Time difference not very huge but it takes less time than previous utility

Reply:
Has this approach been tested on large dataset which was shared by Cisco? Would be good to compare on that set.